filemanager.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2023 qbee.io
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// Manager manages the sync operation.
+type Manager struct {
```

Comment:
Let's call it `FileManager`, as it lives in the shared package where this name could potentially collide with other (future) managers. Then in the comment's where you type `filemanager`, it would be great to write `FileManager` when you refer to this type or `file manager` when you refer to the file manager API/UI in the qbee app.

It seems like I minor thing, but my IDE looks like an x-mass tree with all the spelling warnings ;)

Reply:
Also, the comment could reflect the main difference from just calling the file manager API, which is the fact that it handles the batch operations. Perhaps:
`FileManager provides implementation of common batch operations inside qbee's file manager.`

Reply:
So I left this comment:

>  in the comment's where you type filemanager, it would be great to write FileManager when you refer to this type or file manager when you refer to the file manager API/UI in the qbee app.
> It seems like I minor thing, but my IDE looks like an x-mass tree with all the spelling warnings ;)

But I still see `filemanager` in a lot of comments. I would really appreciate if this could be improved. 
Using `FileManager` when referring to a type, allows me to navigate to that type in my IDE.

Reply:
Sorry, I missed that. Fixed now.